app/uk/gov/hmrc/nisp/utils/Breadcrumb.scala

```diff
@@ -50,10 +50,14 @@ trait Breadcrumb {
     initialBreadCrumbList ::: items.flatten
   }
 
-  def generateHeaderUrl(hideBreadcrumb: Boolean = false) (implicit request:Request[_], user: NispUser): String = {
-    mainContentHeaderPartialUrl + "?name=" + s"${URLEncoder.encode(user.name.getOrElse(""),"UTF-8")}" + "&" +
-      user.previouslyLoggedInAt.map("lastLogin=" + _.getMillis + "&").getOrElse("") +
-      (if(hideBreadcrumb) "" else buildBreadCrumb(request).map{case (name: String, url: String) => s"item_text=$name&item_url=$url"}.mkString("&")) +
-      "&showBetaBanner=true&deskProToken='NISP'"
+  def generateHeaderUrl(hideBreadcrumb: Boolean = false) (implicit request:Request[_], user: NispUser): String =  {
+
+    val name: String =  if(user.name.isDefined) {"?name=" + s"${URLEncoder.encode(user.name.getOrElse(""), "UTF-8")}"} else "?"
+    val lastLogin =  user.previouslyLoggedInAt.map("lastLogin=" + _.getMillis + "&").getOrElse("")
+    val showBreadcrumb = if (hideBreadcrumb) "" else  { buildBreadCrumb(request).map {
+        case (name: String, url: String) => s"item_text=$name&item_url=$url" }.mkString("&")} +
+              "&showBetaBanner=true&deskProToken='NISP'"
```

Comment:
I'm not sure if the end string belongs in the `showBreadCrumb`, should it be in a variable on it's own?
Or perhaps they could be with the start of the "?" rather than having that in the name variable. What do you think?